### ISSUE TITLE
SAM3-Tracker-MLX

### ISSUE THREAD

**Issue by ryouchinsa** [Blaizzy/mlx-vlm#1271]
[sam3-tracker-mlx](https://github.com/ryouchinsa/sam3-tracker-mlx) is to run the [SAM3 Tracker](https://huggingface.co/facebook/sam3) model faster on macOS devices using MLX.

We improved the [MLX-VLM](https://github.com/Blaizzy/mlx-vlm) SAM3 Tracker code so that point, box, and mask input prompts work correctly. If we use the same image resizing and nomalization functions as [PyTorch Transformers](https://github.com/huggingface/transformers), the result mask, iou_score, and object_score_logits are exactly the same between our MLX implementation and the Pytorch Transformers. 

If you have high spec Apple M Series devices such as M2 - M5, could you tell us the processing time of this code on your device?

### MLX result on our Apple M1 device.

```bash
# load_model: 746.50 ms
# feature_maps: 5056.80 ms
# pred_masks
(1, 1, 1, 288, 288)
array([[[[[-13.5184, -12.7845, -12.2627, ..., -12.71, -11.5591, -11.3816],
          [-12.9192, -15.5912, -15.5127, ..., -14.1778, -14.4347, -11.439],
          [-15.5103, -13.8262, -15.3593, ..., -15.0312, -14.0051, -14.776],
          ...,
          [-12.6695, -17.5009, -15.1329, ..., -11.4361, -10.9833, -6.98978],
          [-14.5389, -13.0723, -16.0342, ..., -3.04374, -8.40482, -8.00219],
          [-10.421, -13.6726, -10.0483, ..., -8.5702, -5.33487, -7.004]]]]], dtype=float32)
# iou_scores
(1, 1, 1)
array([[[0.618768]]], dtype=float32)
# object_score_logits
(1, 1, 1)
array([[[21.2302]]], dtype=float32)
# predict: 48.70 ms
```

### PyTorch Transformers result on our Apple M1 device.

```bash
# load_model: 2586.00 ms
# processor: 58.41 ms
# pred_masks
torch.Size([1, 1, 1, 288, 288])
tensor([[[[[-13.5184, -12.7845, -12.2627,  ..., -12.7099, -11.5591,
            -11.3816],
           [-12.9191, -15.5912, -15.5127,  ..., -14.1778, -14.4347,
            -11.4390],
           [-15.5103, -13.8262, -15.3593,  ..., -15.0312, -14.0051,
            -14.7760],
           ...,
           [-12.6695, -17.5009, -15.1329,  ..., -11.4362, -10.9834,
             -6.9898],
           [-14.5389, -13.0723, -16.0342,  ...,  -3.0438,  -8.4049,
             -8.0022],
           [-10.4210, -13.6725, -10.0482,  ...,  -8.5702,  -5.3349,
             -7.0040]]]]])
# iou_scores
torch.Size([1, 1, 1])
tensor([[[0.6188]]])
# object_score_logits
torch.Size([1, 1, 1])
tensor([[[21.2302]]])
# predict: 13674.61 ms
```

**Comment by Lazarus-931** [comment]
hi @ryouchinsa,

I tested it out, got these on you repo:

<img width="697" height="178" alt="Image" src="https://github.com/user-attachments/assets/b1c68c73-c9f0-40aa-a935-fdb502f147f9" />
